woudc_data_registry/processing.py

```diff
@@ -1020,9 +1030,9 @@ def check_data_record(self):
         version = self.extcsv.extcsv['DATA_GENERATION']['Version']
         dg_valueline = self.extcsv.line_num('DATA_GENERATION') + 2
 
-        components = self.data_record.data_record_id.split(':')
-        components[-1] = '%.%'
-        identifier_pattern = ':'.join(components)
+        id_components = data_record.data_record_id.split(':')
+        id_components[-1] = '%.%'
```

Comment:
What's `%.%` ?

Reply:
`%.%` is PSQL for the regular expression `".\.."`. Two wildcards separated by a dot. It matches any floating-point version number, and so you end up getting a previously existing version of the data record.

Reply:
can we make `registry.query_by_pattern` do this logic?  Having `%.%` which is an SQL-ism is better isolated in `woudc_data_registry.registry.query_by_pattern`.